### PR TITLE
[bug][python] Fix task switch error when default branch not defined last

### LINKED ISSUE #11605: [BUG] [Python Switch Task] The default branch will report an error if it is not defined last
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

image

```shell
Traceback (most recent call last):
  File "src/pydolphinscheduler/examples/task_switch_example.py", line 50, in <module>
    pd.submit()
  File "/home/lucky/WhaleOps/dolphinscheduler/dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/core/process_definition.py", line 410, in submit
    json.dumps(self.task_definition_json),
  File "/home/lucky/WhaleOps/dolphinscheduler/dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/core/process_definition.py", line 247, in task_definition_json
    return [task.get_define() for task in self.tasks.values()]
  File "/home/lucky/WhaleOps/dolphinscheduler/dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/core/process_definition.py", line 247, in <listcomp>
    return [task.get_define() for task in self.tasks.values()]
  File "/home/lucky/WhaleOps/dolphinscheduler/dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/models/base.py", line 69, in get_define
    content = self.get_define_custom(camel_attr, self._DEFINE_ATTR)
  File "/home/lucky/WhaleOps/dolphinscheduler/dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/models/base.py", line 56, in get_define_custom
    val = getattr(self, attr, None)
  File "/home/lucky/WhaleOps/dolphinscheduler/dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/tasks/switch.py", line 162, in task_params
    params["switchResult"] = self.condition.get_define()
  File "/home/lucky/WhaleOps/dolphinscheduler/dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/tasks/switch.py", line 128, in get_define
    self.set_define_attr()
  File "/home/lucky/WhaleOps/dolphinscheduler/dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/tasks/switch.py", line 112, in set_define_attr
    raise PyDSParamException(
pydolphinscheduler.exceptions.PyDSParamException: Task Switch's parameter only support exactly one default branch.
```

### What you expected to happen

Should be successful

### How to reproduce

Do not define the default branch at the end

### Anything else

_No response_

### Version

dev

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by jieguangzhou**
## Purpose of the pull request

close: #11605

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

**Comment by jieguangzhou** [comment]
@zhongjiajie PTAL, thanks

File: dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/tasks/switch.py [hunk 1]
```diff
@@ -107,7 +107,7 @@ def set_define_attr(self) -> None:
                         num_branch_default += 1
                     elif isinstance(condition, Branch):
                         result.append(condition.get_define())
-                else:
+                elif num_branch_default >= 1 and isinstance(condition, Default):
```
Comment (zhongjiajie) [inline dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/tasks/switch.py]: This code is a little hard to read, maybe we should also change the code range, raise error ASAP when we meet it. It is my bad about this code

```py
    def set_define_attr(self) -> None:
        """Set attribute to function :func:`get_define`.

        It is a wrapper for both `And` and `Or` operator.
        """
        result = []
        num_branch_default = 0
        for condition in self.args:
            if not isinstance(condition, SwitchBranch):
                raise PyDSParamException(
                    "Task Switch's parameter only support SwitchBranch but got %s.",
                    type(condition),
                )
            # Default number branch checker
            if num_branch_default >= 1 and isinstance(condition, Default):
                raise PyDSParamException(
                    "Task Switch's parameter only support SwitchBranch but got %s.",
                    type(condition),
                )
            if isinstance(condition, Default):
                self._DEFINE_ATTR.add("next_node")
                setattr(self, "next_node", condition.next_node)
                num_branch_default += 1
            elif isinstance(condition, Branch):
                result.append(condition.get_define())
        # Handle switch default branch, default value is `""` if not provide.
        if num_branch_default == 0:
            self._DEFINE_ATTR.add("next_node")
            setattr(self, "next_node", "")
        setattr(self, "depend_task_list", result)
```

Comment (zhongjiajie) [reply dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/tasks/switch.py]: WDYT @jieguangzhou

Comment (jieguangzhou) [reply dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/tasks/switch.py]: Yes, it's best to rearrange the order. I've already resubmitted it

**Review by zhongjiajie — APPROVED** [review]

**Pull request merged**